x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_details/common_components.cy.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { deleteRuleFromDetailsPage } from '../../../tasks/alerts_detection_rules';
+import {
+  CUSTOM_RULES_BTN,
+  RULES_MANAGEMENT_TABLE,
+  RULES_ROW,
+} from '../../../screens/alerts_detection_rules';
+import { createRule } from '../../../tasks/api_calls/rules';
+import { ruleDetailsUrl } from '../../../urls/navigation';
+import { getDetails } from '../../../tasks/rule_details';
+import { ruleFields } from '../../../data/detection_engine';
+import { getTimeline } from '../../../objects/timeline';
+import { getExistingRule, getNewRule } from '../../../objects/rule';
+
+import {
+  ABOUT_DETAILS,
+  ABOUT_INVESTIGATION_NOTES,
+  ABOUT_RULE_DESCRIPTION,
+  ADDITIONAL_LOOK_BACK_DETAILS,
+  CUSTOM_QUERY_DETAILS,
+  DEFINITION_DETAILS,
+  FALSE_POSITIVES_DETAILS,
+  INDEX_PATTERNS_DETAILS,
+  INVESTIGATION_NOTES_MARKDOWN,
+  INVESTIGATION_NOTES_TOGGLE,
+  REFERENCE_URLS_DETAILS,
+  removeExternalLinkText,
+  RISK_SCORE_DETAILS,
+  RULE_NAME_HEADER,
+  RULE_SWITCH,
+  RULE_TYPE_DETAILS,
+  RUNS_EVERY_DETAILS,
+  SCHEDULE_DETAILS,
+  SEVERITY_DETAILS,
+  TAGS_DETAILS,
+  THREAT_SUBTECHNIQUE,
+  THREAT_TACTIC,
+  THREAT_TECHNIQUE,
+  TIMELINE_TEMPLATE_DETAILS,
+} from '../../../screens/rule_details';
+
+import { createTimeline } from '../../../tasks/api_calls/timelines';
+import { cleanKibana, deleteAlertsAndRules, deleteConnectors } from '../../../tasks/common';
+import { login, visitWithoutDateRange } from '../../../tasks/login';
+
+// This test is meant to test all common aspects of the rule details page that should function
+// the same regardless of rule type. For any rule type specific functionalities, please include
+// them in the relevant /rule_details/[RULE_TYPE].cy.ts test.
+describe('Common components', { tags: ['@ess', '@serverless'] }, () => {
```

Comment:
nit: maybe renaming to common scenarios would slightly better fit to its purpose?
Since we test rule types scenarios, not isolated components